src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -63,16 +62,71 @@ case class ArrayIteratorTriplet(calcLength: Code[Unit], length: Option[Code[Int]
 }
 
 private class Emit(
-  fb: FunctionBuilder[_],
-  mb: StagedBitSet,
+  mb: MethodBuilder,
   tAggInOpt: Option[TAggregable],
   nSpecialArguments: Int) {
 
+  private val maxBytecodeSizeTarget: Int = 2048
+  private val opSize: Int = 20
+
+  def chunkIRs(irs: Seq[IR]): Array[Seq[IR]] = {
+    val chunkBounds = getChunkBounds(irs.map(_.size * opSize))
+    Array.tabulate(chunkBounds.length - 1)(i => irs.slice(chunkBounds(i), chunkBounds(i + 1)))
+  }
+
+  def getChunkBounds(sizes: Seq[Int]): Array[Int] = {
```

Comment:
Let's also have a test that gives examples of how we expect this method to operate.